### PR TITLE
Fixed logic for which Razor scaffolder gets invoked.

### PR DESCRIPTION
**Pull request by seancpeters**
When generating all pages, the templateName is "All". Initial code had danticipated it being null.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/RazorPage/RazorPageGenerator.cs [hunk 1]
```diff
@@ -51,16 +51,18 @@ public class RazorPageGenerator : CommonGeneratorBase, ICodeGenerator
 
             if (string.IsNullOrEmpty(razorPageGeneratorModel.ViewName))
             {
-                // when this happens, we want to generate all the views
+                throw new ArgumentException(MessageStrings.ViewNameRequired);
```
Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/RazorPage/RazorPageGenerator.cs]: This argument will still be null for the `All` template.

Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/RazorPage/RazorPageGenerator.cs]: Looks like this change is compatible with the `2.0.0` packages, so it should be fine. 
Just need to move the null check for `ViewName` to after the `TemplateName` check and then 
:shipit:

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs [hunk 2]
```diff
@@ -43,13 +43,22 @@ private static void Execute(string[] args, ConsoleLogger logger)
             var noBuild = app.Option("--no-build", "", CommandOptionType.NoValue);
             var simMode = app.Option("--simulation-mode", Resources.SimulationModeOptionDesc, CommandOptionType.NoValue);
 
-#if DEBUG
-            if (args.Contains("--debug", StringComparer.OrdinalIgnoreCase))
+// the for reals version
+//
+//#if DEBUG
+//            if (args.Contains("--debug", StringComparer.OrdinalIgnoreCase))
```
Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs]: Looks like we need to evolve this condition to support debugging when testing using VS. 
Maybe along with the parameter, look for an environment variable?

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs [hunk 3]
```diff
@@ -43,13 +43,22 @@ private static void Execute(string[] args, ConsoleLogger logger)
...
+//            {
+//                Console.WriteLine($"Attach a debugger to processID: {System.Diagnostics.Process.GetCurrentProcess().Id} and hit enter.");
+//                Console.ReadKey();
+//            }
+//#endif
+//
+
+// for testing via VS
+            while (!System.Diagnostics.Debugger.IsAttached)
```
Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs]: Remove debug code.

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs (hunk 2 above)
Comment (seancpeters) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs]: Yeah, not sure what exactly is the right way... I can experiment.

File: src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs (hunk 3 above)
Comment (seancpeters) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.Design/Program.cs]: Whoops, yep, will do now.

File: src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/RazorPage/EmptyRazorPageScaffolder.cs [hunk 4]
```diff
@@ -35,7 +35,6 @@ public class EmptyRazorPageScaffolder : RazorPageScaffolderBase
 
             if (string.IsNullOrEmpty(razorGeneratorModel.ViewName))
             {
-                // TODO: make a separate message resource string for this (currently setup using the one for VIEW)
                 throw new ArgumentException(MessageStrings.ViewNameRequired);
```
Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/RazorPage/EmptyRazorPageScaffolder.cs]: We should change these to `Razor Page Name is required.`

**Pull request merged**